aws/resource_aws_s3_bucket_metric.go

```diff
@@ -81,21 +84,26 @@ func resourceAwsS3BucketMetricPut(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	log.Printf("[DEBUG] Putting metric configuration: %s", input)
-	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+	err := resource.Retry(waiter.PropagationTimeout, func() *resource.RetryError {
 		_, err := conn.PutBucketMetricsConfiguration(input)
+
+		if tfawserr.ErrCodeEquals(err, s3.ErrCodeNoSuchBucket) {
+			return resource.RetryableError(err)
+		}
+
 		if err != nil {
-			if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") {
-				return resource.RetryableError(err)
-			}
 			return resource.NonRetryableError(err)
 		}
+
 		return nil
 	})
-	if isResourceTimeoutError(err) {
+
+	if tfresource.TimedOut(err) {
 		_, err = conn.PutBucketMetricsConfiguration(input)
 	}
+
 	if err != nil {
-		return fmt.Errorf("Error putting S3 metric configuration: %s", err)
+		return fmt.Errorf("error putting S3 Bucket Metric Configuration: %w", err)
```

Comment:
Nit: i noticed below `Metrics` is used instead of `Metric` so for consistency maybe use the plural here as well..unless there's an intended diff?

Reply:
Great catch. I will make them consistent to `Metrics` in a followup commit here. 👍  One day we'll create and use a Go type that will centralize all this information for error reporting, etc. 😍 

```go
&resource{
  FriendlyName: "S3 Bucket Metrics Configuration",
  // ...
}
```